Source/loadsave.cpp

```diff
@@ -1990,7 +2038,17 @@ void SaveGameData()
 	file.WriteLE<uint8_t>(chrflag ? 1 : 0);
 	file.WriteBE<int32_t>(ActiveMonsterCount);
 	file.WriteBE<int32_t>(ActiveItemCount);
-	file.WriteBE<int32_t>(ActiveMissileCount);
+	int missileCountOldFormat = 0;
+	int missileCountAdditional = 0;
+	if (leveltype != DTYPE_TOWN) {
+		for (auto &missileWithId : Missiles) {
+			if (missileWithId.first < MaxMissilesForSaveGame)
+				missileCountOldFormat += 1;
+			else
+				missileCountAdditional += 1;
+		}

```

Comment:
This potentially leaves the vanilla save file with far less than 125 active missiles. For example if 150 get spawned at once and the 120 oldest are deleted (leaving 30 active with IDs 121-150), the vanilla save file contains 5 active missiles, the additional file contains 25.

If you handle this like we currently do with `Objects` you can set up the vanilla save file in a way that active missiles start from index 0 and go in ascending order. Then you can just write the first 125 missiles here and whatever overflow there is in the additional file.

Also there's a potential bug with town portal missiles and the corresponding light source. I think it'll end up with an orphaned light source if the corresponding missile gets written to the additional save file, but lights are buggy in general 😅

Reply:
You are right if we don't want to support stable id's (see [discussion](https://github.com/diasurgical/devilutionX/pull/3957#discussion_r786291804)).
If we want to support stable id's this is one downside of it. 😉

Reply:
Fixed. 😉